snapcraft/plugins/v2/_ros.py

```diff
@@ -0,0 +1,180 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import abc
+import os
+import pathlib
+import subprocess
+import sys
+from typing import Dict, List, Set
+
+import click
+from catkin_pkg import packages as catkin_packages
+
+from snapcraft.internal.repo import Repo
+from snapcraft.plugins.v1._ros.rosdep import _parse_rosdep_resolve_dependencies
+from snapcraft.plugins.v2 import PluginV2
+
+
+class RosPlugin(PluginV2):
+    """Base class for ROS-related plugins. Not intended for use by end users."""
+
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        return {
+            "python3-rosdep",
+        }
+
+    def get_build_environment(self) -> Dict[str, str]:
+        return {
+            "ROS_PYTHON_VERSION": "3",
+        }
+
+    def _get_workspace_activation_commands(self) -> List[str]:
+        """
+        Return a list of commands source a ROS workspace.
```

Comment:
This line should be next to """ (https://www.python.org/dev/peps/pep-0257/)
```suggestion
        """Return a list of commands source a ROS workspace.
```

Reply:
Fixed.